File: demo/components/ui.window.showText.ts
import ui from "interactive-script-js";
import { longText } from "./constants";

async function showTextDemo() {
    ui.success("Open text in new window demo.").fontSize(18).print();

    ui.log("");
    ui.text("You can open text in new window.");
    await ui.dialog.buttons(["Show me"]);

    ui.window.showText({
        text: longText,
        language: "plaintext",
    });

    ui.log("");
    ui.success("End of demo.");
}

showTextDemo()
    .catch(e => console.error(e))
    .finally(() => process.exit(0));

File: python_demo/components/ui_window_show_text.py
import asyncio
from interactive_script_py import ui, styled_text
from utils import longText

async def window_show_text_demo():
    ui.success("Open text in new window demo.").font_size(18).print()

    ui.log("")
    ui.text("You can open text in new window.")
    await ui.dialog.buttons(["Show me"])

    ui.window.show_text({
        "text": longText,
        "language": "plaintext",
    })

    ui.log("")
    ui.success("End of demo.")

if __name__ == "__main__":
    asyncio.run(window_show_text_demo())